### PR TITLE
PhpUnit 8 Migration - AdminNotification

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description (*)
This PR updates unit tests in AdminNotification module + contains framework fix from https://github.com/magento/magento2/pull/27519

Note: 
We have 2 warnings that can't be fixed because these new methods doesn't exists in phpunit 6:
```
1) Magento\AdminNotification\Test\Unit\Model\System\Message\CacheOutdatedTest::testGetText
Using assertContains() with string haystacks is deprecated and will not be supported in PHPUnit 9. Refactor your test to use assertStringContainsString() or assertStringContainsStringIgnoringCase() instead.

2) Magento\AdminNotification\Test\Unit\Model\System\Message\Media\Synchronization\ErrorTest::testGetText
Using assertContains() with string haystacks is deprecated and will not be supported in PHPUnit 9. Refactor your test to use assertStringContainsString() or assertStringContainsStringIgnoringCase() instead.
```

### Fixed Issues (if relevant)

1. https://github.com/magento/magento2/issues/27500

### Manual testing scenarios (*)
1. Run tests in phpunit 6 (currently in 2.4-develop) using `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/AdminNotification/`
2. Install phpunit 8, run tests using `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/AdminNotification/`

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php]: There's missing reference to original class type.

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php]: There's missing reference to original class type.

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php]: There's missing reference to original class type.

File: app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Test/Unit/Model/System/Message/SecurityTest.php]: There's missing reference to original class type.

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
In future we can change the scope for Test properties from `protected` to `private`.
I'm not sure if it's for purpose or accidentally, but you've added changes from other PR.

You can put in the description that this PR would work after merging another (however the changes introduced in files related to AdminNotification are backward compatible, so there's no dependency)

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@lbajsarowicz 
I fixed comments that you left. Why I put there one commit from framework - just to make these pull requests independent, as the same commit is there in al PRs - there will not be any conflicts between them and all of these PRs will actually work

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7243](https://jira.corp.magento.com/browse/ENGCOM-7243) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by ihor-sviziev** [comment]
@lenaorobei seems like we have false positive in static test

**Comment by lenaorobei** [comment]
@ihor-sviziev looking into it.

File: lib/internal/Magento/Framework/TestFramework/Unit/Listener/GarbageCleanup.php
Comment (lenaorobei) [inline lib/internal/Magento/Framework/TestFramework/Unit/Listener/GarbageCleanup.php]: This is correct finding.
`20 | ERROR | [x] If the @inheritdoc not inline it shouldn’t have braces`

File: lib/internal/Magento/Framework/TestFramework/Unit/Listener/GarbageCleanup.php
Comment (lenaorobei) [inline lib/internal/Magento/Framework/TestFramework/Unit/Listener/GarbageCleanup.php:22]: Missing blank line.

**Comment by lenaorobei** [comment]
Yes, the 3-rd finding looks like false-positive, but it should disappear when first 2 will be fixed.

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
@lenaorobei @lbajsarowicz Could you approve the latest changes?
Thanks!

**Review by lbajsarowicz — APPROVED** [review]
Yes, I reviewed them before and was waiting for tests to pass.
We have all green, so my approval is there.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7243](https://jira.corp.magento.com/browse/ENGCOM-7243) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
PHPUnit v8
php 7.3

**Before:**
Screenshot from 2020-04-02 16-47-39

**After:**
Screenshot from 2020-04-02 16-38-14

**Pull request merged**
